pkg/skaffold/schema/v1beta9/upgrade.go

```diff
@@ -17,18 +17,26 @@ limitations under the License.
 package v1beta9
 
 import (
+	"strings"
+
 	next "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/util"
 	pkgutil "github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	incompatibleSyncWarning = `the semantics of the sync rules has changed, the folder structure is not flattened anymore but preserved, the likely impacted patterns in your skaffold yaml are: %s`
```

Comment:
Might be nice to provide a link to the docs here so people can easily find the new sync rules